### ISSUE TITLE
Should scripts have context?

### ISSUE THREAD

**Issue by alexksikes** [elastic/elasticsearch#13375]
It would be nice if scripts could have some kind of context in which all the necessary imports would be performed as well as some global variables initialization. This is necessary, for example, when we want to evaluate a machine learning model in say an aggregation. The context would be loading the model together with its vectorizer only once, so that it could be evaluated repeatedly while performing the aggregation. Currently, it seems that the only way to achieve this would be to pass in as parameters to the script the serialized model, yet the model would still need to be deserialized each time. Otherwise the only other possibility would be to move to native scripts.

**Comment by jpountz** [comment]
I am a bit worried that we have the temptation to go in two opposite directions with scripts:
 1. making them simple, safe and fast, potentially at the cost of flexibility (#13084)
 2. making them more flexible (eg. this issue)

If you ask me, I think 1 is more important. 2 might not be completely incompatible with 1 but I think we should do 1 before 2 if we don't want to end up in a situation where this new scripting language becomes the go-to solution for scripting but it can not support new options that we would introduce in this issue.

**Comment by clintongormley** [comment]
Closing as won't fix
